c2corg_api/views/document.py

```diff
@@ -17,20 +17,69 @@ class DocumentRest(object):
     def __init__(self, request):
         self.request = request
 
-    def _collection_get(self, clazz, schema, adapt_schema=None):
-        documents = DBSession. \
-            query(clazz). \
+    def _paginate_offset(self, clazz, schema, adapt_schema):
+        validated = self.request.validated
+        offset = validated['offset'] if 'offset' in validated else 0
+        limit = validated['limit'] if 'limit' in validated else 30
+        total = validated['total'] if 'total' in validated else None
+
+        base_query = DBSession. \
+            query(clazz)
+
+        documents = base_query. \
+            options(joinedload(getattr(clazz, 'locales'))). \
+            order_by(clazz.document_id.desc()). \
+            slice(offset, offset + limit). \
+            all()
+        set_available_cultures(documents)
+
+        total = base_query.count() if total is None else total
```

Comment:
But how can you be sure that no document was added since the last request? The client does not know.

Instead we could cache the query on the server-side. What do you think?